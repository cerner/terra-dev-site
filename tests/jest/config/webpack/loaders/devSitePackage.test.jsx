import 'regenerator-runtime/runtime';

const path = require('path');
const { runLoaders } = require('loader-runner');

describe('devSitePackage loader test', () => {
  it('when both repository url and repository directory are specified', (done) => {
    runLoaders({
      resource: path.resolve(__dirname, './pretendPackageOne.json'),
      loaders: [
        path.resolve(__dirname, '../../../../../config/webpack/loaders/devSitePackage'),
      ],
    }, (err, result) => {
      if (err) return done(err);
      expect(result.result[0]).toMatchSnapshot();
      return done();
    });
  });

  it('when only repository url is specified', (done) => {
    runLoaders({
      resource: path.resolve(__dirname, './pretendPackageTwo.json'),
      loaders: [
        path.resolve(__dirname, '../../../../../config/webpack/loaders/devSitePackage'),
      ],
    }, (err, result) => {
      if (err) return done(err);
      expect(result.result[0]).toMatchSnapshot();
      return done();
    });
  });

  it('when only repository directory is specified', (done) => {
    runLoaders({
      resource: path.resolve(__dirname, './pretendPackageThree.json'),
      loaders: [
        path.resolve(__dirname, '../../../../../config/webpack/loaders/devSitePackage'),
      ],
    }, (err, result) => {
      if (err) return done(err);
      expect(result.result[0]).toMatchSnapshot();
      return done();
    });
  });

  it('when repository attribute is not specified', (done) => {
    runLoaders({
      resource: path.resolve(__dirname, './pretendPackageFour.json'),
      loaders: [
        path.resolve(__dirname, '../../../../../config/webpack/loaders/devSitePackage'),
      ],
    }, (err, result) => {
      if (err) return done(err);
      expect(result.result[0]).toMatchSnapshot();
      return done();
    });
  });

  it('when url is specified without git+ prefix and directory starts with front slash', (done) => {
    runLoaders({
      resource: path.resolve(__dirname, './pretendPackageFive.json'),
      loaders: [
        path.resolve(__dirname, '../../../../../config/webpack/loaders/devSitePackage'),
      ],
    }, (err, result) => {
      if (err) return done(err);
      expect(result.result[0]).toMatchSnapshot();
      return done();
    });
  });

  it('when the repository url is not a git url', (done) => {
    runLoaders({
      resource: path.resolve(__dirname, './pretendPackageSix.json'),
      loaders: [
        path.resolve(__dirname, '../../../../../config/webpack/loaders/devSitePackage'),
      ],
    }, (err, result) => {
      if (err) return done(err);
      expect(result.result[0]).toMatchSnapshot();
      return done();
    });
  });
});
