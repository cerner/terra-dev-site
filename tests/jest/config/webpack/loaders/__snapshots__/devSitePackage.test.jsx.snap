// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`devSitePackage loader test when both repository url and repository directory are specified 1`] = `
"import React from 'react';
import Badges from '@cerner/terra-dev-site/lib/loader-components/_Badges';

export const Badge = ({ url }) => (
      <Badges
        src=\\"https://github.com/cerner/terra-dev-site/tree/main/src/loader-components\\"
        name=\\"terra-dev-site\\"
        version=\\"6.2.0\\"
        url={url}
      />
    );"
`;

exports[`devSitePackage loader test when only repository directory is specified 1`] = `
"import React from 'react';
import Badges from '@cerner/terra-dev-site/lib/loader-components/_Badges';

export const Badge = ({ url }) => (
      <Badges
        name=\\"terra-dev-site\\"
        version=\\"6.2.0\\"
        url={url}
      />
    );"
`;

exports[`devSitePackage loader test when only repository url is specified 1`] = `
"import React from 'react';
import Badges from '@cerner/terra-dev-site/lib/loader-components/_Badges';

export const Badge = ({ url }) => (
      <Badges
        src=\\"https://github.com/cerner/terra-dev-site\\"
        name=\\"terra-dev-site\\"
        version=\\"6.2.0\\"
        url={url}
      />
    );"
`;

exports[`devSitePackage loader test when repository attribute is not specified 1`] = `
"import React from 'react';
import Badges from '@cerner/terra-dev-site/lib/loader-components/_Badges';

export const Badge = ({ url }) => (
      <Badges
        name=\\"terra-dev-site\\"
        version=\\"6.2.0\\"
        url={url}
      />
    );"
`;

exports[`devSitePackage loader test when the repository url is not a git url 1`] = `
"import React from 'react';
import Badges from '@cerner/terra-dev-site/lib/loader-components/_Badges';

export const Badge = ({ url }) => (
      <Badges
        src=\\"https://v8.googlecode.com/svn/trunk/\\"
        name=\\"terra-dev-site\\"
        version=\\"6.2.0\\"
        url={url}
      />
    );"
`;

exports[`devSitePackage loader test when url is specified without git+ prefix and directory starts with front slash 1`] = `
"import React from 'react';
import Badges from '@cerner/terra-dev-site/lib/loader-components/_Badges';

export const Badge = ({ url }) => (
      <Badges
        src=\\"https://github.com/cerner/terra-dev-site/tree/main/src/loader-components\\"
        name=\\"terra-dev-site\\"
        version=\\"6.2.0\\"
        url={url}
      />
    );"
`;
